### PR TITLE
refactoring 1.0

### PR DESCRIPTION
**Pull request by kof**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -19,92 +52,23 @@ export default class FiniteList extends Component {
...
     this.onScrollStopDebounced = debounce(::this.onScrollStop, 30)
   }
 
-  componentWillReceiveProps(nextProps) {
-    this.setState(this.createState(nextProps))
-  }
-
   componentDidMount() {
-    this.setState({node: ReactDOM.findDOMNode(this)})
+    if (!this.node) this.forceUpdate()
```
Comment (tyv) [inline src/index.js:64]: Think needs a comment

File: src/index.js [hunk 2]
```diff
@@ -144,4 +102,64 @@ export default class FiniteList extends Component {
...
+
+  checkSensor(item) {
+    const index = findIndex(this.props.items, _item => _item === item)
+    this.refs[`sensor-${index}`].check()
+  }
+
+  renderItems() {
+    const {items} = this.props
+
+    // We need theis DOM node for visibility sensor.
```
Comment (tyv) [inline src/index.js]: spelling `theis`

Comment (tyv) [inline src/index.js]: also comment is bad for who didn't know what sensor is it about

Comment (tyv) [inline src/index.js]: ah now see.

**Comment by tyv** [comment]
@kof only few very small comments above

**Comment by kof** [comment]
@tyv done

**Comment by tyv** [comment]
@kof ping me if any changes or you will remove `[wip]`

**Pull request merged**